src/main/java/net/smartcosmos/dao/relationships/impl/RelationshipPersistenceService.java

```diff
@@ -32,237 +39,284 @@
 
     @Autowired
-    public RelationshipPersistenceService(RelationshipRepository RelationshipRepository,
-            ConversionService conversionService) {
+    public RelationshipPersistenceService(
+        RelationshipRepository RelationshipRepository,
+        ConversionService conversionService) {
         this.relationshipRepository = RelationshipRepository;
         this.conversionService = conversionService;
     }
 
     // region Create
-
     @Override
-    public Optional<RelationshipResponse> create(String tenantUrn, RelationshipCreate createRelationship) throws ConstraintViolationException {
-        // TODO: Implement method
-        throw new UnsupportedOperationException("Not implemented yet");
-    }
-
-    // endregion
+    public Optional<RelationshipResponse> create(String tenantUrn, RelationshipCreate createRelationship) {
 
-    // region Delete
+        if (alreadyExists(tenantUrn, createRelationship)) {
+            return Optional.empty();
+        }
+        RelationshipEntity entity = null;
+        try {
+            UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
+            entity = conversionService.convert(createRelationship, RelationshipEntity.class);
+            entity.setTenantId(tenantId);
+            entity = persist(entity);
+
+        } catch (Exception e) {
+            exceptionLogger("RelationshipPersistenceService:create", e, tenantUrn, createRelationship);
+            throw e;
+        }
+        return Optional.ofNullable(conversionService.convert(entity, RelationshipResponse.class));
+    }
+    //endregion
 
+    // region delete
     @Override
     public List<RelationshipResponse> delete(String tenantUrn, String urn) {
 
-        UUID accountId = UuidUtil.getUuidFromUrn(tenantUrn);
+        UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
 
-        List<RelationshipEntity> deleteList = new ArrayList<>();
         try {
             UUID uuid = UuidUtil.getUuidFromUrn(urn);
-            deleteList = relationshipRepository.deleteByTenantIdAndId(accountId, uuid);
-        } catch (IllegalArgumentException e) {
-            // empty list will be returned anyway
-            log.warn("Illegal URN submitted: %s by account %s", urn, tenantUrn);
-        }
+            return convertList(relationshipRepository.deleteByTenantIdAndId(tenantId, uuid));
 
-        return convert(deleteList);
+        } catch (Exception e) {
+            exceptionLogger("RelationshipPersistenceService:delete", e, "tenantUrn: " + tenantUrn, "urn: " + urn);
+        }
+        return new ArrayList<>();
     }
-
     // endregion
 
     // region Find specific / by URN
-
     @Override
     public Optional<RelationshipResponse> findByUrn(String tenantUrn, String urn) {
 
-        UUID accountId = UuidUtil.getUuidFromUrn(tenantUrn);
-
+        UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
         Optional<RelationshipEntity> entity = Optional.empty();
```

Comment:
Another situation where an object is being created and may be thrown away.  Move this create  to the `catch()` block